### PR TITLE
ZEN-21540: add warning for user when use zencatalog '-f'

### PR DESCRIPTION
**Pull request by rderkach**

### CODE REVIEW

File: src/zenoss/toolbox/zencatalogscan.py [hunk 1]
```diff
@@ -363,6 +363,20 @@ def main():
...
           (time.strftime("%Y-%m-%d %H:%M:%S"), scriptName, scriptVersion, logFileName)
 
+    # Give user option to stop if he hadn't run findposkeyerror, zenrelationscan, zodbscan
+    if cliOptions["fix"]:
+        while True:
+            user_answer = raw_input("You should run findposkeyerror, "
+                                    "zenrelationscan, zodbscan before "
+                                    "running zencatalogscan -f cause "
+                                    "it may damage your data. If you "
+                                    "done it print [y] to continue "
```
Comment (glujan) [inline src/zenoss/toolbox/zencatalogscan.py]: Ain't printing

**Comment by jstanley23** [comment]
Can a flag/argument be added to allow a user to skip this? I think its a great reminder, but we automate the use of these scripts and i would hate to have to script out a response for this input

**Comment by rderkach** [comment]
@jstanley23 I agree with you. I will work on it tomorrow.

File: src/zenoss/toolbox/zencatalogscan.py (hunk 1 above)
Comment (rderkach) [inline src/zenoss/toolbox/zencatalogscan.py]: Fixed.

File: src/zenoss/toolbox/zencatalogscan.py [hunk 2]
```diff
@@ -363,6 +366,19 @@ def main():
     print "\n[%s] Initializing %s v%s (detailed log at %s)" % \
           (time.strftime("%Y-%m-%d %H:%M:%S"), scriptName, scriptVersion, logFileName)
 
+    # Give user option to stop if he hadn't run findposkeyerror, zenrelationscan, zodbscan
+    if cliOptions["fix"] and not cliOptions["force_yes"]:
```
Comment (glujan) [inline src/zenoss/toolbox/zencatalogscan.py]: `cliOptions["force_yes"]` should be checked to decide if we want user to give his answer directly or just assume he responded with `y`. Know if we use this flag from command line, `fix` will never happen.

File: src/zenoss/toolbox/zencatalogscan.py [hunk 3]
```diff
@@ -354,6 +354,9 @@ def main():
                         help="output all supported catalogs")
     parser.add_argument("-c", "--catalog", action="store", default="",
                         help="only scan/fix specified catalog")
+    parser.add_argument("--force-yes", action="store_true", default=False,
+                        help="continue without prompting, relavant"
+                             "to '-f' option, but it may damage your data")
```
Comment (glujan) [inline src/zenoss/toolbox/zencatalogscan.py]: Please make this help text more general (like it's assuming response `y` to _all_ prompts) or rename it to something more relevant for using only with `--fix` argument (maybe `--force-fix`).

Comment (rderkach) [inline src/zenoss/toolbox/zencatalogscan.py]: Fixed.

**Pull request merged**